### PR TITLE
Fix/refactor hydration

### PR DESCRIPTION
**Pull request by ADavidson02**
## Is this a fix or a feature?
Fix

## What is the change?
Hydration.js

## What does it fix?

Refactor methods for the hydration class 

## Where should the reviewer start?
hydration.js line 2

## How should this be tested?
There are chai test for this class in the hydration-test.js

@Shawntru 
@adavidson02
@sarah01123

### CODE REVIEW

File: test/hydration-test.js [hunk 1]
```diff
@@ -39,139 +51,148 @@ describe('Hydration', () => {
...
-      email: 'rainbowD1@hotmail.com',
-      strideLength: 3.8,
-      dailyStepGoal: 7000,
-      friends: [1, 2, 3],
-    });
-    const users = [user3, user4];
-    const userRepo = new UserRepo(users);
-    console.log('HELOOO', hydration.calculateRandomWeekOunces('2018/02/01', 4, userRepo));
-    expect(hydration.calculateRandomWeekOunces('2019/09/18', 4, userRepo)[0]).to.eql('2019/09/18: 40');
+  // it('should find  by day for that days week', () => {
```
Comment (saraho1123) [inline test/hydration-test.js:169]: Is this a test we want to move to UserRepo?

File: src/Hydration.js [hunk 2]
```diff
@@ -1,25 +1,58 @@
...
+  constructor(hydrationData, user) {
+    this.userId = user.id;
+    this.hydrationData = this.getHydrationData(hydrationData);
   }
-
-  calculateAverageOunces(id) {
-    const perDayUserHydration = this.hydrationData.filter((data) => id === data.userID);
-    return perDayUserHydration.reduce((sumSoFar, data) => sumSoFar += data.numOunces, 0) / perDayUserHydration.length;
+  
+  getHydrationData(hydrationData) {
```
Comment (saraho1123) [inline src/Hydration.js:7]: We probably need to leave an ID's related to the raw data to capital ID, even though we agreed in our project we would use Id. Hopefully this makes sense!

File: test/hydration-test.js [hunk 3]
```diff
@@ -4,32 +4,44 @@ import UserRepo from '../src/User-repo';
...
    
+      friends: [
 16,
 2, 6]
   
+    }
+    user = new User(userData)
     hydrationData = [{
-      userID: 1,
+      userId: 1,
```
Comment (saraho1123) [inline test/hydration-test.js:24]: We want this userId to be userID, as we want it to be like our data files that we are working with.

File: src/Hydration.js [hunk 4]
```diff
@@ -1,25 +1,58 @@
...
+        return wholeWeek
+      }, [])
+      return weekHydration
+    }
+  
+    findGivenWeek(date) {
+      let sorted = this.hydrationData.sort((a, b) => {
+        return Date.parse(a.date) - Date.parse(b.date)
+      })
+        let endDate = sorted.findIndex(data =>{
```
Comment (Shawntru) [inline src/Hydration.js:46]: Looks good, let's make sure we are following our style guide indenting protocols (Line 11, calculateDailyOunces() on line 24, and line 35 and 39 inside calculateFirstWeekOunces())

**Review by saraho1123 — COMMENTED** [review]
This looks great to me! Just need to change back the Id's that are data.

File: src/Hydration.js (hunk 2 above)
Comment (ADavidson02) [reply src/Hydration.js:7]: Got it will change that to an upper ID

**Review by Shawntru — APPROVED** [review]
Looks good, left comment on line 46

**Pull request merged**